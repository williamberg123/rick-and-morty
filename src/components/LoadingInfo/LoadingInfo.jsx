import React, { useContext } from 'react';

import AppContext from '../../AppContext';

import './LoadingInfo.css';

export default function LoadingInfo(){
    const { allReturnedData, filteredData } = useContext(AppContext);
    
    const totalToLoad = allReturnedData.length;
    const totalLoaded = filteredData.length;

    return (
        <div className="LoadingInfo">
            <div className="Loading-total">Total: <span>{totalToLoad}</span></div>
            <div className="Loading-loaded-count">Carregado at√© o momento: <span>{totalLoaded}</span></div>
        </div>
    );
}
