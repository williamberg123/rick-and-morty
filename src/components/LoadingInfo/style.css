.LoadingInfo {
    width: 100%;
    margin: 0 10px;
    display: flex;
    flex-direction: column;
    font-weight: bold;
}

.LoadingInfo span {
    color: red;
}