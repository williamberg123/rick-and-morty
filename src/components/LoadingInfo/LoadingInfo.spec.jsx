import React from 'react';
import { screen, render } from '@testing-library/react';
import LoadingInfo from './LoadingInfo';

describe('<LoadingInfo />', () => {
    it('should render informations correctly', () => {
        expect.assertions(4);
        render(<LoadingInfo total={826} loaded={20} />);

        expect(screen.getByText(/Total:/i)).toBeInTheDocument();
        expect(screen.getByText(/Total:/i)).toHaveClass('Loading-total');

        expect(screen.getByText(/Carregado até o momento:/i)).toBeInTheDocument();
        expect(screen.getByText(/Carregado até o momento:/i)).toHaveClass('Loading-loaded-count');
    });

    it('should match snapshot', () => {
        render(<LoadingInfo total={826} loaded={20} />);

        expect(screen.getByText(/Total:/i).parentElement)
            .toMatchSnapshot();
    });
});

// Testado