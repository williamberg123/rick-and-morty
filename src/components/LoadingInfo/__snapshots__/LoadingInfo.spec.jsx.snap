// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingInfo /> should match snapshot 1`] = `
<div
  class="LoadingInfo"
>
  <div
    class="Loading-total"
  >
    Total: 
    <span>
      826
    </span>
  </div>
  <div
    class="Loading-loaded-count"
  >
    Carregado at√© o momento: 
    <span>
      20
    </span>
  </div>
</div>
`;
